gossip/c_event_callbacks.go

```diff
@@ -143,13 +143,21 @@ func (s *Service) SwitchEpochTo(newEpoch idx.Epoch) error {
 	}
 	s.engineMu.Lock()
 	defer s.engineMu.Unlock()
+	s.blockProcWg.Wait()
 	if newEpoch == s.store.GetEpoch() {
 		return errSameEpoch
 	}
 	s.store.SetBlockEpochState(*bs, *es)
 	s.switchEpochTo(newEpoch)
-	_ = s.store.EvmSnapshotAt(common.Hash(bs.FinalizedStateRoot))
-	return nil
+	err := s.engine.Reset(newEpoch, es.Validators)
+	if err != nil {
+		return err
+	}
+	err = s.store.EvmSnapshotAt(common.Hash(bs.FinalizedStateRoot))
```

Comment:
nitpick: it might not the issue but it sound more logically if we move the snapshot generating before switching to new epoch